subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/DynamicVersionResolver.java

```diff
@@ -114,7 +116,7 @@ private RepositoryChainModuleResolution findLatestModule(List<RepositoryResolveS
 
         // A first pass to do local resolves only
         RepositoryChainModuleResolution best = findLatestModule(queue, failures, missing);
-        if (!failures.isEmpty()) {
+        if (hasCriticalFailure(failures)) {
```

Comment:
⭕️ Consider asking `blacklister.isBlacklisted()` instead of leaking an internal blacklister implementation method here.

Reply:
Let's not do that. Instead, the Blacklister and the "continue-on-failure" logic should use the same mechanism to determine if a failure is 'critical'. Ideally, we'd push this detection down to the source where we wrap the underlying failure, so these 2 things could just say `if (failure.isCritical()) ...`.

For now, I'll just move the static `isCriticaFailurel()` and `hasCriticalFailures()` into a separate, shared type.